pkg/proxy/proxy.go

```diff
@@ -142,6 +149,181 @@ func (p *Proxy) allocatePort() (uint16, error) {
 
 var gcOnce sync.Once
 
+// localEndpointInfo fills the access log with the local endpoint info.
+func localEndpointInfo(r Redirect, info *accesslog.EndpointInfo) {
+	source := r.getSource()
+	source.RLock()
+	info.ID = source.GetID()
+	info.IPv4 = source.GetIPv4Address()
+	info.IPv6 = source.GetIPv6Address()
+	info.Labels = source.GetLabels()
+	info.LabelsSHA256 = source.GetLabelsSHA()
+	info.Identity = uint64(source.GetIdentity())
+	source.RUnlock()
+}
+
+// GetSourceInfo resolves source information
+// using source identity and fills the access log.
+func GetSourceInfo(RemoteAddr string,
```

Comment:
This can still be made private

Reply:
sure. my concepts on private functions in a golang package are a bit shaky :) Will do

Reply:
done